### PR TITLE
[KYUUBI #4388] Limit the max rows for get nextRowSet api

### PR DESCRIPTION
**Pull request by lightning-L**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by lightning-L** [comment]
@turboFei

File: docs/deployment/settings.md [hunk 1]
```diff
@@ -440,6 +440,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
+| kyuubi.operation.rest.fetch.max.rows            | 5000                                                                            | Max rows limit for getting result row set api. If the max rows specified by client-side is larger than the limit, request will fail directly.                                                                                                                                                                                                                                                                                                                                                            | int      | 1.7.0 |
```
Comment (turboFei) [inline docs/deployment/settings.md:443]: not only for rest call, we need also limit the jdbc

File: docs/deployment/settings.md [hunk 2]
```diff
@@ -431,6 +431,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
+| kyuubi.operation.client.fetch.max.rows          | 5000                                                                            | Max rows limit for getting result row set operation. If the max rows specified by client-side is larger than the limit, request will fail directly.                                                                                                                                                                                                                                                                                                                                                      | int      | 1.7.0 |
```
Comment (turboFei) [inline docs/deployment/settings.md:434]: Refer the style liks:
```
kyuubi.server.limit.connections.per.user
``


how about:

```
kyuubi.server.limit.client.fetch.max.rows
```

Comment (turboFei) [reply docs/deployment/settings.md:434]: refer the config:

  val SERVER_LIMIT_CONNECTIONS_PER_USER_IPADDRESS: OptionalConfigEntry[Int] =
    buildConf("kyuubi.server.limit.connections.per.user.ipaddress")
      .doc("Maximum kyuubi server connections per user:ipaddress combination." +
        " Any user-ipaddress exceeding this limit will not be allowed to connect.")
      .version("1.6.0")
      .serverOnly
      .intConf
      .createOptional

File: docs/deployment/settings.md [hunk 3]
```diff
@@ -455,6 +455,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
 | kyuubi.server.batch.limit.connections.per.user.ipaddress | &lt;undefined&gt; | Maximum kyuubi server batch connections per user:ipaddress combination. Any user-ipaddress exceeding this limit will not be allowed to connect.                                                                                                | int      | 1.7.0 |
 | kyuubi.server.info.provider                              | ENGINE            | The server information provider name, some clients may rely on this information to check the server compatibilities and functionalities. <li>SERVER: Return Kyuubi server information.</li> <li>ENGINE: Return Kyuubi engine information.</li> | string   | 1.6.1 |
+| kyuubi.server.limit.client.fetch.max.rows                | 5000              | Max rows limit for getting result row set operation. If the max rows specified by client-side is larger than the limit, request will fail directly.                                                                                            | int      | 1.7.0 |
```
Comment (turboFei) [inline docs/deployment/settings.md:458]:
```
kyuubi.server.limit.connections.per.user.ipaddress
kyuubi.server.batch.limit.connections.per.user.ipaddress
kyuubi.server.limit.client.fetch.max.rows   
```


should we rename the config `kyuubi.server.batch.limit.connections.per.user.ipaddress` to `kyuubi.server.limit.batch.connections.per.user.ipaddress` ? @pan3793

Comment (turboFei) [reply docs/deployment/settings.md:458]: raised #4398

File: kyuubi-common/src/main/scala/org/apache/kyuubi/service/AbstractBackendService.scala [hunk 4]
```diff
@@ -201,6 +201,11 @@ abstract class AbstractBackendService(name: String)
       orientation: FetchOrientation,
       maxRows: Int,
       fetchLog: Boolean): TRowSet = {
+    val maxRowsLimit = conf.get(KyuubiConf.SERVER_LIMIT_CLIENT_FETCH_MAX_ROWS)
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/service/AbstractBackendService.scala:204]: how about using a lazy val.

```
private lazy val maxRowsLimit = conf.get(KyuubiConf.SERVER_LIMIT_CLIENT_FETCH_MAX_ROWS)
```

**Comment by turboFei** [comment]
Af think twice， I think that we can set the default value to optional， to do not break the original behavior.

**Review by turboFei — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**